packages/alchemy/src/react/components/auth/hooks/usePasskeyVerify.ts

```diff
@@ -0,0 +1,29 @@
+import { useAuthenticate } from "../../../hooks/useAuthenticate.js";
+import { useAuthContext } from "../context.js";
+
+/**
+ * Internal UI component hook used to complete passkey verification.
+ *
+ * This is used to login with a passkey, not create a new passkey
+ *
+ * @returns an authenticate function to do passkey verification and a boolean indicating if the operation is pending
+ */
+export const usePasskeyVerify = () => {
+  const { setAuthStep } = useAuthContext();
+  const { authenticate, isPending } = useAuthenticate({
+    onMutate: () => {
+      setAuthStep({ type: "passkey_verify" });
+    },
+    onError: (err) => {
+      setAuthStep({ type: "passkey_verify", error: err });
+    },
+    onSuccess: () => {
+      setAuthStep({ type: "complete" });
+    },
+  });
+
+  return {
+    isPending,
+    authenticate: () => authenticate({ type: "passkey", createNew: false }),
```

Comment:
As an improvement, we might want to think about using `useCallback` functions we return to the SDK user. Some devs may have their apps set up to attempt to minimize re-renders using `memo` and returning a new function instance on each call to `usePasskeyVerify` could mess them up.

This comment assumes that `useAuthenticate` is returning the same `authenticate` instance every time. It's possible that implementing this suggestion codebase-wide would end up being fairly involved, and doesn't need to be done in this PR.

Reply:
ah yea `useAuthenticate` does return the same instance (as long as react-query does). 

I can update this authenticate method to be wrapped in `useCallback` based on that assumption